engines/bops_admin/app/controllers/bops_admin/submissions_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module BopsAdmin
+  class SubmissionsController < ApplicationController
+    before_action :set_submissions, only: %i[index]
+
+    rescue_from Pagy::OverflowError do
+      redirect_to submissions_path
+    end
```

Comment:
Probably fine to do it this way, but there's also an optional extra in pagy itself to configure the overflow behaviour: https://ddnexus.github.io/pagy/docs/extras/overflow/ (although it looks like redirecting to the first page isn't actually an option in that case).